src/test/java/org/cactoos/http/io/BlockingSocketServer.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.http.io;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Useful object for tests that needs to timeout on connect.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @since 0.1
+ */
+public final class BlockingSocketServer implements Closeable {
+
+    /**
+     * The server.
+     */
+    private final ServerSocket server;
+
+    /**
+     * Sockets needed to block the server.
+     */
+    private final List<Socket> blockers;
+
+    /**
+     * Ctor.
+     * @throws IOException If the server cannot be created
+     */
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public BlockingSocketServer() throws IOException {
+        this.server = new ServerSocket(0, 1);
+        this.blockers = new LinkedList<>();
+        this.blockers.add(new Socket(this.address(), this.port()));
+        this.blockers.add(new Socket(this.address(), this.port()));
+    }
+
+    /**
+     * Retrieve the server address.
+     * @return Server address.
+     */
+    public InetAddress address() {
+        return this.server.getInetAddress();
+    }
+
+    /**
+     * Retrieve the server port.
+     * @return A positive port number.
+     */
+    public int port() {
+        return this.server.getLocalPort();
+    }
+
+    @Override
+    public void close() throws IOException {
```

Comment:
@victornoel couldn't cactoos classes be used for this method to follow a more object-oriented approach?

Reply:
@scristalli do you have something specific in mind? I don't see which abstraction can be useful

Reply:
@victornoel I must confess it does seem a little hard to refactor completely. Maybe the use of `And` can help to remove the `if`, and the `for` can be translated to some map operation? Just a suggestion, I'm open to discussion.

Reply:
@scristalli here you have it (careful for what you wish for ;). I had "fun" doing it, I'm not 100% convinced it's the way to go, but maybe some abstractions can be extracted from that and brought back to cactoos, so it was worth playing with it :)